components/suggest/src/rs.rs

```diff
@@ -520,6 +530,81 @@ pub(crate) struct DownloadedFakespotSuggestion {
     pub url: String,
 }
 
+/// An exposure suggestion to ingest from an attachment
+#[derive(Clone, Debug, Deserialize)]
+pub(crate) struct DownloadedExposureSuggestion {
+    pub exposure_suggestion_type: String,
+    keywords: Vec<FullOrPrefixKeywords<String>>,
+}
+
+impl DownloadedExposureSuggestion {
+    /// Iterate over all keywords for this suggestion. Iteration may contain
+    /// duplicate keywords depending on the structure of the data, so do not
+    /// assume keywords are unique. Duplicates are not filtered out because
+    /// doing so would require O(number of keywords) space, and the number of
+    /// keywords can be very large. If you are inserting into the store, rely on
+    /// uniqueness constraints and use `INSERT OR IGNORE`.
+    pub fn keywords(&self) -> impl Iterator<Item = String> + '_ {
+        self.keywords.iter().flat_map(|e| e.keywords())
+    }
+}
+
+/// A single full keyword or a `(prefix, suffixes)` tuple representing multiple
+/// prefix keywords. Prefix keywords are enumerated by appending to `prefix`
+/// each possible prefix of each suffix, including the full suffix. The prefix
+/// is also enumerated by itself. Examples:
+//
+/// `FullOrPrefixKeywords::Full("some full keyword")`
+/// => "some full keyword"
+///
+/// `FullOrPrefixKeywords::Prefix(("sug", vec!["gest", "arplum"]))`
+/// => "sug"
+///    "sugg"
+///    "sugge"
+///    "sugges"
+///    "suggest"
+///    "suga"
+///    "sugar"
+///    "sugarp"
+///    "sugarpl"
+///    "sugarplu"
+///    "sugarplum"
```

Comment:
I wonder if this should be using FTS search rather than explicit keywords, since we might be using FTS more in the future.  I really don't know though, what's your thoughts?

Reply:
We need to have pretty tight control over keywords and matching, mostly due to legal reasons. Every time we run a potential exposure experiment, we have to clear the keywords with legal. There may be a little wiggle room for fuzzy matching, but things like stemming and arbitrary prefix matching can alter the set of "actual" keywords too dramatically. Other than that legal reason, there's also a lot of value in having all experiment stakeholders on the same page w/r/t the keywords being tested, and it becomes harder to explain to PMs, QA, and external teams (who come up with the keywords) what exactly we're testing if the matching is too fuzzy.

That said, I do think there's probably a nice middle ground that uses FTS, and ideally we wouldn't require external teams to come up with huge lists of keyword phrases, many of which are variations on each other. I'd like to start with the simpler strategy in this commit and think about FTS for a follow-up.